activerecord/lib/active_record/connection_adapters/postgresql/database_statements.rb

```diff
@@ -170,8 +132,61 @@ def cancel_any_running_query
           rescue PG::Error
           end
 
+          def perform_query(raw_connection, sql, binds, type_casted_binds, prepare:, notification_payload:)
+            update_typemap_for_default_timezone
+            result = if prepare
+              begin
+                stmt_key = prepare_statement(sql, binds, raw_connection)
+                notification_payload[:statement_name] = stmt_key
+                raw_connection.exec_prepared(stmt_key, type_casted_binds)
+              rescue PG::FeatureNotSupported => error
+                if is_cached_plan_failure?(error)
+                  # Nothing we can do if we are in a transaction because all commands
+                  # will raise InFailedSQLTransaction
+                  if in_transaction?
+                    raise PreparedStatementCacheExpired.new(error.message, connection_pool: @pool)
+                  else
+                    @lock.synchronize do
+                      # outside of transactions we can simply flush this query and retry
+                      @statements.delete sql_key(sql)
+                    end
+                    retry
+                  end
+                end
+
+                raise
+              end
+            elsif without_prepared_statement?(binds)
+              raw_connection.async_exec(sql)
```

Comment:
This change is discarding the `binds` for queries where `prepare` is false, which is currently breaking `good_job` in many places.  Is this change intentional or a regression?  I believe binds should be included regardless of whether prepare is true but I am no expert!

Reply:
Also this branch isn't entrered unless binds are empty / nil, so I don't see how it could break anything.

Reply:
The error appeared [here](https://github.com/bensheldon/good_job/blob/a61fc09cdc06ba89a9615da90778459f13a837a8/app/charts/good_job/scheduled_by_queue_chart.rb#L33) for us.  Currently, this branch is entered as long as `!prepared_statements` returns true (regardless of how binds is set)

```
def without_prepared_statement?(binds)
  !prepared_statements || binds.nil? || binds.empty?
end
```

Reply:
https://github.com/rails/rails/pull/52580